### PR TITLE
Autoselect exact match

### PR DESCRIPTION
**Pull request by annasirunian**
Pasted reactions and compounds which have the exact match with the result from metanetx service (mnx and all ids from the annotation) are now autoselected.

TODO:
Highlight autocompletes without matches

### CODE REVIEW

File: src/components/AutocompleteMnxReaction.vue [hunk 1]
```diff
@@ -167,7 +211,7 @@ export default Vue.extend({
             if (isReactionFound) {
               searchResultsInTheModel.push(reaction);
             } else {
-              reaction.namespace = "metanetx.chemical";
+              reaction.namespace = "metanetx.reaction";
```
Comment (lukaw3d) [inline src/components/AutocompleteMnxReaction.vue]: Good catch

File: src/components/AutocompleteMnxMetabolite.vue [hunk 2]
```diff
@@ -178,6 +224,22 @@ export default Vue.extend({
...
+            const firstResult = this.searchResults[0];
+            const metaboliteIds = new Set(
+              Object.values(firstResult.annotation).flat()
+            );
+            metaboliteIds.add(firstResult.mnx_id);
+            if (
+              metaboliteIds.has(this.searchQuery) ||
+              this.searchQuery === firstResult.name
+            ) {
+              this.selectedValue = firstResult;
```
Comment (lukaw3d) [inline src/components/AutocompleteMnxMetabolite.vue]: Not needed?

File: src/components/AutocompleteMnxMetabolite.vue [hunk 3]
```diff
@@ -178,6 +224,22 @@ export default Vue.extend({
             );
           }
           this.searchResults.push(...searchResultsNotInTheModel);
+          // If pasted metabolite has the exact match with the first result
+          // from metanetx service, it should be autoselected
+          if (this.searchQuery === this.forceSearchQuery) {
+            const firstResult = this.searchResults[0];
+            const metaboliteIds = new Set(
+              Object.values(firstResult.annotation).flat()
```
[the remark this answers is not in the document]
Comment (kvikshaug) [reply src/components/AutocompleteMnxMetabolite.vue]: Ah, good catch. Let's use the lodash version to ensure compatibility for IE users

File: src/components/AutocompleteMnxMetabolite.vue (hunk 2 above)
Comment (annasirunian) [reply src/components/AutocompleteMnxMetabolite.vue]: 👍

**Pull request merged**